main/ui.js

```diff
@@ -86,10 +83,30 @@ module.exports = async function (ctx) {
   // that were added to keep app running when the UI is closed.
   // (Without this, the app would run forever and/or fail to update)
   app.on('before-quit', () => {
-    ipcMain.removeListener(ipcMainEvents.ACTIVITY_LOGGED, onNewActivity)
+    stopIpcEventForwarding()
```

Comment:
is this actually necessary at all?

Reply:
Yes. When we don't unsubscribe our IPC event listeners, they get triggered while the app is shutting down. Our handler forwards the event to `ui`, which has already been destroyed. Unhandled rejection is triggered, the app shutdown does not complete, and the app keeps running.

Reply:
Done in https://github.com/filecoin-project/filecoin-station/pull/136/commits/66cab6f94a376971d952e936a010fa34a3e8b309.

I find the `request` API slightly more ergonomic than the `fetch` API, but I guess there is more value in using a standard Web API that more people are familiar with.

Reply:
> Unhandled rejection is triggered, the app shutdown does not complete, and the app keeps running.

Ok! It's better to not have an unhandled rejection then